spec/beaker-answers/pe_conf_spec.rb

```diff
@@ -97,4 +178,103 @@
+      split_basic_hosts[1]['platform'] = 'el-6-x86_64'
+      split_basic_hosts[2]['roles'] = ['dashboard','agent']
+      split_basic_hosts[2]['platform'] = 'el-6-x86_64'
+      split_basic_hosts[3]['roles'] = ['agent']
+      split_basic_hosts[3]['platform'] = 'el-6-x86_64'
+      split_basic_hosts
+    end
+
+    it 'returns master in monolithic' do
+      peconf = BeakerAnswers::PeConf.new(mono_hosts, "1.0")
+      expect(peconf.send(:the_host_with_role, 'master')).to eq(mono_basic_hosts[0])
+    end
+
+    it 'returns database in monolithic' do
+      peconf = BeakerAnswers::PeConf.new(mono_hosts, "1.0")
+      expect(peconf.send(:the_host_with_role, 'database')).to eq(mono_basic_hosts[0])
+    end
+
+    it 'returns dashboard in monolithic' do
+      peconf = BeakerAnswers::PeConf.new(mono_hosts, "1.0")
+      expect(peconf.send(:the_host_with_role, 'dashboard')).to eq(mono_basic_hosts[0])
+    end
+
+    it 'returns master in split' do
+      peconf = BeakerAnswers::PeConf.new(split_hosts, "1.0")
+      expect(peconf.send(:the_host_with_role, 'master')).to eq(split_basic_hosts[0])
+    end
+
+    it 'returns database in split' do
+      peconf = BeakerAnswers::PeConf.new(split_hosts, "1.0")
+      expect(peconf.send(:the_host_with_role, 'database')).to eq(split_basic_hosts[1])
+    end
+
+    it 'returns dashboard in split' do
+      peconf = BeakerAnswers::PeConf.new(split_hosts, "1.0")
+      expect(peconf.send(:the_host_with_role, 'dashboard')).to eq(split_basic_hosts[2])
+    end
+
+    it 'raises an error if multiple hosts have master role' do
+      mono_hosts[1]['roles'] = ['master','agent']
+      peconf = BeakerAnswers::PeConf.new(mono_hosts, "1.0")
+      expect{ peconf.send(:the_host_with_role, 'master') }.to raise_error(ArgumentError)
+    end
+
+    it 'raises an error if multiple hosts have database role' do
+      mono_hosts[1]['roles'] = ['database','agent']
+      peconf = BeakerAnswers::PeConf.new(mono_hosts, "1.0")
+      expect{ peconf.send(:the_host_with_role, 'database') }.to raise_error(ArgumentError)
+    end
+
+    it 'raises an error if multiple hosts have dashboard role' do
+      mono_hosts[1]['roles'] = ['dashboard','agent']
+      peconf = BeakerAnswers::PeConf.new(mono_hosts, "1.0")
+      expect{ peconf.send(:the_host_with_role, 'dashboard') }.to raise_error(ArgumentError)
+    end
+
+    it 'raises no error if multiple hosts have database role and flag is passed in' do
+      mono_hosts[1]['roles'] = ['database','agent']
+      peconf = BeakerAnswers::PeConf.new(mono_hosts, "1.0")
+      expect(peconf.send(:the_host_with_role, 'database', raise_error=false)).to eq(mono_basic_hosts[0])
+    end
+
+    it 'returns postgres node in monolithic if pe_postgres is set as a role' do
+      mono_hosts[1]['roles'] = ['pe_postgres','agent']
+      peconf = BeakerAnswers::PeConf.new(mono_hosts, "1.0")
+      expect(peconf.send(:the_host_with_role, 'pe_postgres', raise_error=false)).to eq(mono_basic_hosts[1])
```

Comment:
Should `raise_error` be passed in here? We're not expecting a duplicate role in this test, are we?

Reply:
Good point, thanks.